src/test/java/com/rakovets/course/java/core/practice/string/StringUtilTest.java

```diff
@@ -0,0 +1,289 @@
+        String actualString = StringUtil.removeSpacesAroundCastUpperCase(str1);
+
+        // THAT
+        Assertions.assertEquals(expectedSymbol, actualString);
+    }
+
+    static Stream<Arguments> extractionSubstring() {
+        return Stream.of(
+                Arguments.of("ll", "Hello  ", 2, 4),
+                Arguments.of("l", " HeLlo", 4, 5),
+                Arguments.of("lati", "Relationship", 2, 6)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("extractionSubstring")
+    void extractionSubstring(String expectedSymbol, String str1, int nSymbol, int mSymbol) {
+        // GIVEN
+
+        // WHEN
+        String actualString = StringUtil.extractionSubstring(str1, nSymbol, mSymbol);
+
+        // THAT
+        Assertions.assertEquals(expectedSymbol, actualString);
+    }
+
+    static Stream<Arguments>happyEmoji() {
+        return Stream.of(
+                Arguments.of("Hello:)  ", "Hello:(  ", "(", ")"),
+                Arguments.of(" HeLlo)", " HeLlo(", "(", ")"),
+                Arguments.of(":):):)", ":(:(:(", "(", ")")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("happyEmoji")
+    void happyEmoji(String expectedSymbol, String str1, String previousSymbol, String modernSymbol) {
+        // GIVEN
+
+        // WHEN
+        String actualString = StringUtil.happyEmoji(str1, previousSymbol, modernSymbol);
+
+        // THAT
+        Assertions.assertEquals(expectedSymbol, actualString);
+    }
+
+    static Stream<Arguments>startEndWith() {
+        return Stream.of(
+                Arguments.of("true", "Hi", "Hi"),
+                Arguments.of("false", "hi !", " Java"),
+                Arguments.of("false", "!Hi", "java")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("startEndWith")
+    void startEndWith(boolean expectedSymbol, String text, String word) {
+        // GIVEN
+
+        // WHEN
+        boolean actualString = StringUtil.startEndWith(text, word);
+
+        // THAT
+        Assertions.assertEquals(expectedSymbol, actualString);
+    }
+
+    static Stream<Arguments>vowels() {
+        return Stream.of(
+                Arguments.of("2", "HEllo!"),
+                Arguments.of("6", "qwertyuiop[]';lkjhgfdsazxcvbnm,./"),
+                Arguments.of("5", "ololoOo")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("vowels")
+    void vowels(int expectedSymbol, String str1) {
+        // GIVEN
+
+        // WHEN
+        int actualString = StringUtil.vowels(str1);
+
+        // THAT
+        Assertions.assertEquals(expectedSymbol, actualString);
+    }
+
+    static Stream<Arguments>punctuations() {
+        return Stream.of(
+                Arguments.of("1", "Hello!"),
+                Arguments.of("2", "qwertyuiop[]';lkjhgfdsazxcvbnm,./"),
+                Arguments.of("6", ",o.l?o!!.looo")
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("punctuations")
+    void punctuations(int expectedSymbol, String str1) {
```

Comment:
"Violation: 'nameMethodTest' was not used for test-method naming
Solution: Rename method
"